fdbserver/SimpleConfigDatabaseNode.actor.cpp

```diff
@@ -0,0 +1,493 @@
+
+Key versionedAnnotationKey(Version version) {
+	ASSERT_GE(version, 0);
+	return BinaryWriter::toValue(bigEndian64(version), IncludeVersion()).withPrefix(annotationKeys.begin);
+}
+
+Version getVersionFromVersionedAnnotationKey(KeyRef versionedAnnotationKey) {
+	return fromBigEndian64(BinaryReader::fromStringRef<uint64_t>(
+	    versionedAnnotationKey.removePrefix(annotationKeys.begin), IncludeVersion()));
+}
+
+Key versionedMutationKey(Version version, uint32_t index) {
+	ASSERT_GE(version, 0);
+	BinaryWriter bw(IncludeVersion());
+	bw << bigEndian64(version);
+	bw << bigEndian32(index);
+	return bw.toValue().withPrefix(mutationKeys.begin);
+}
+
+Version getVersionFromVersionedMutationKey(KeyRef versionedMutationKey) {
+	uint64_t bigEndianResult;
+	ASSERT(versionedMutationKey.startsWith(mutationKeys.begin));
+	BinaryReader br(versionedMutationKey.removePrefix(mutationKeys.begin), IncludeVersion());
+	br >> bigEndianResult;
+	return fromBigEndian64(bigEndianResult);
+}
+
+} //namespace
+
+TEST_CASE("/fdbserver/ConfigDB/SimpleConfigDatabaseNode/Internal/versionedMutationKeys") {
+	std::vector<Key> keys;
+	for (Version version = 0; version < 1000; ++version) {
+		for (int index = 0; index < 5; ++index) {
+			keys.push_back(versionedMutationKey(version, index));
+		}
+	}
+	for (int i = 0; i < 5000; ++i) {
+		ASSERT(getVersionFromVersionedMutationKey(keys[i]) == i / 5);
+	}
+	return Void();
+}
+
+TEST_CASE("/fdbserver/ConfigDB/SimpleConfigDatabaseNode/Internal/versionedMutationKeyOrdering") {
+	Standalone<VectorRef<KeyRef>> keys;
+	for (Version version = 0; version < 1000; ++version) {
+		for (auto index = 0; index < 5; ++index) {
+			keys.push_back_deep(keys.arena(), versionedMutationKey(version, index));
+		}
+	}
+	for (auto index = 0; index < 1000; ++index) {
+		keys.push_back_deep(keys.arena(), versionedMutationKey(1000, index));
+	}
+	ASSERT(std::is_sorted(keys.begin(), keys.end()));
+	return Void();
+}
+
+class SimpleConfigDatabaseNodeImpl {
+	UID id;
+	OnDemandStore kvStore;
+	std::map<std::string, std::string> config;
+
+	CounterCollection cc;
+
+	// Follower counters
+	Counter compactRequests;
+	Counter successfulChangeRequests;
+	Counter failedChangeRequests;
+	Counter snapshotRequests;
+
+	// Transaction counters
+	Counter successfulCommits;
+	Counter failedCommits;
+	Counter setMutations;
+	Counter clearMutations;
+	Counter getValueRequests;
+	Counter newVersionRequests;
+	Future<Void> logger;
+
+	ACTOR static Future<Version> getLiveTransactionVersion(SimpleConfigDatabaseNodeImpl *self) {
+		Optional<Value> value = wait(self->kvStore->readValue(liveTransactionVersionKey));
+		state Version liveTransactionVersion = 0;
+		if (value.present()) {
+			liveTransactionVersion = BinaryReader::fromStringRef<Version>(value.get(), IncludeVersion());
+		} else {
+			self->kvStore->set(KeyValueRef(liveTransactionVersionKey, BinaryWriter::toValue(liveTransactionVersion, IncludeVersion())));
```

Comment:
You're right, this isn't strictly necessary. I only set in this case to be consistent, and because it might make reading and debugging the `kvStore` contents easier.